README.md

```diff
@@ -8,21 +8,76 @@
 
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[Mistral AI](https://chat.mistral.ai/chat?q=) is a research lab focused on developing the best open-source AI models. It provides developers and businesses with powerful [APIs](https://docs.mistral.ai/api/) and tools to build innovative applications using both free and commercial large language models.
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Mistral AI Connector, you must have access to the Mistral AI API through a Mistral AI account and an active API key. If you do not have a Mistral AI account, you can sign up for one [here](https://mistral.ai/)
+
+#### Create a Mistral AI API key
+
+1. Visit the Mistral AI platform, head to the Mistral AI console dashboard, and sign up to get started.
+
+2. Navigate to the API Keys panel.
+
+3. Choose a plan based on your requirements.
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-mistral/main/docs/setup/resources/choose-plan.png alt="Mistral AI Platform" style="width: 70%;">
+
+4. Proceed to create a new API key.
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-mistral/main/docs/setup/resources/create-key.png alt="Mistral AI Platform" style="width: 70%;">
+
+5. Enter the necessary details as prompted and click on Create new key.
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-mistral/main/docs/setup/resources/details-panel.png alt="Mistral AI Platform" style="width: 70%;">
+
+6. Copy the API key and store it securely
+<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-mistral/main/docs/setup/resources/copy-key.png alt="Mistral AI Platform" style="width: 70%;">
```

Comment:
Shall we omit the HTML stylings in the markdown? These image widths are now being handled from the Ballerina Central.